cas-server-webapp-actions/src/main/java/org/jasig/cas/web/flow/InitialFlowSetupAction.java

```diff
@@ -67,13 +65,23 @@
     @Override
     protected Event doExecute(final RequestContext context) throws Exception {
         final HttpServletRequest request = WebUtils.getHttpServletRequest(context);
-        if (!this.pathPopulated) {
-            final String contextPath = context.getExternalContext().getContextPath();
-            final String cookiePath = StringUtils.hasText(contextPath) ? contextPath + '/' : "/";
-            logger.info("Setting path for cookies to: {} ", cookiePath);
+
+        final String contextPath = context.getExternalContext().getContextPath();
+        final String cookiePath = StringUtils.isNotBlank(contextPath) ? contextPath + '/' : "/";
+
+        if (StringUtils.isBlank(warnCookieGenerator.getCookiePath())) {
```

Comment:
Yes, if there is no context path defined. In otherwise, if you have deployed CAS at root and you have not set the cookie path to root explicitly, it will be set to root.